REGRESSION(252693@main): ASSERTION FAILED: results.size() == results.capacity() in Interpreter::getStackTrace

dcrousso opened the pull request:
#### 700b13b162339206b1308a3c774caf82c6676f91
<pre>
REGRESSION(252693@main): ASSERTION FAILED: results.size() == results.capacity() in Interpreter::getStackTrace
<a href="https://bugs.webkit.org/show_bug.cgi?id=243082">https://bugs.webkit.org/show_bug.cgi?id=243082</a>

Reviewed by Yusuke Suzuki.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::GetStackTraceFunctor::GetStackTraceFunctor):
(JSC::GetStackTraceFunctor::operator() const):
(JSC::Interpreter::getStackTrace):
Also skip anyting that isn&apos;t `ImplementationVisibility::Public` when counting the number of frames
to capture in the stack trace (whereas 252693@main only did that when actually capturing the frames).
Drive-by: Clean up some of the logic to make it easier to read.

Canonical link: <a href="https://commits.webkit.org/252722@main">https://commits.webkit.org/252722@main</a>
</pre>

Constellation commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -435,20 +431,26 @@ void Interpreter::getStackTrace(JSCell* owner, Vector<StackFrame>& results, size
+                return IterationStatus::Continue;
+        }
+
+        if (++visitedFrames <= maxStackSize)
```
How about,
```
if (++visitedFrames < maxStackSize)
```
And just use `visitedFrames` instead of `std::min(visitedFrames, maxStackSize)`?

dcrousso replied:
oh lol good call

Constellation reviewed (approved):
r=me

dcrousso committed:
REGRESSION(252693@main): ASSERTION FAILED: results.size() == results.capacity() in Interpreter::getStackTrace
https://bugs.webkit.org/show_bug.cgi?id=243082

Reviewed by Yusuke Suzuki.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::GetStackTraceFunctor::GetStackTraceFunctor):
(JSC::GetStackTraceFunctor::operator() const):
(JSC::Interpreter::getStackTrace):
Also skip anyting that isn't `ImplementationVisibility::Public` when counting the number of frames
to capture in the stack trace (whereas 252693@main only did that when actually capturing the frames).
Drive-by: Clean up some of the logic to make it easier to read.

Canonical link: https://commits.webkit.org/252722@main
